components/base/sensorcontrolled/movestraight.go

```diff
@@ -43,13 +43,26 @@ func (sb *sensorBase) MoveStraight(
 		}
 		return sb.controlledBase.MoveStraight(ctx, distanceMm, mmPerSec, extra)
 	}
+	if sb.position == nil {
+		sb.logger.CWarn(ctx,
+			"Position reporting sensor not available, controlling using linear velocity only")
+
+		// adjust inputs to ensure errDist is always positive to match the position based implementation
+		if distanceMm < 0 {
+			distanceMm = -distanceMm
+			mmPerSec = -mmPerSec
+		}
```

Comment:
when using a position sensor the distance error always starts at a positive number, so I had to change the inputs for the not-position case to make the behavior match. 

this lets both cases share the same desired velocity logic(which gets passed into the pid controller)